### DISCUSSION TITLE
random error when the car is asleep

### DISCUSSION THREAD

**Discussion by GV78** [tdorssers/TeslaPy#132] [Q&A]
I make this error from time to time:
```
    lat = vehicles[0]['drive_state']['latitude']
  File "/usr/local/lib/python3.6/dist-packages/teslapy/__init__.py", line 591, in __missing__
    raise KeyError(key)
KeyError: 'drive_state'
```

I think it's when the car has trouble waking up from standby, but I'm not sure...

I've tried using try before the command, but I don't catch the error and the program simply quits...
How can I deal with this error?

here's a snippet of my code when the error occurs:
```
    vehicles = tesla.vehicle_list()
    vehicles[0].sync_wake_up()
    vehicles[0].get_vehicle_data()
    lat = vehicles[0]['drive_state']['latitude']
    print("Lat", lat)
    lon = vehicles[0]['drive_state']['longitude']
    print("Lon", lon)
```

**Comment by tdorssers** [comment]
Please add `print(vehicles[0].available())` before the sync_wake_up line. If it prints True, then the vehicle has gone to sleep in the previous minute.

**Reply by GV78** [reply]
Thanks, I'll give it a try

If the vehicle has gone to sleep in the previous minute, what should I do?

**Comment by GV78** [comment]
I didn't understand what `vehicles[0].available()` returns.
I still have `True`, and for 5 days the script has worked, and last night it crashed again:

```
True
Traceback (most recent call last):
  File "./tesla-charge.py", line 46, in <module>
    lat = vehicles[0]['drive_state']['latitude']
  File "/usr/local/lib/python3.6/dist-packages/teslapy/__init__.py", line 591, in __missing__
    raise KeyError(key)
KeyError: 'drive_state'
```
What can I do to track the error and do a retry?
